Description: silence erroneous RLIMIT_NUMFDS-related log messages
 Fixes setrlimit(RLIMIT_NUMFDS) handling to be less obnoxious and
 not barf error messages to syslog incorrectly, nor log nonsense
 if getrlimit(RLIMIT_NUMFDS) failed.
Author: Henrique de Moraes Holschuh <hmh@debian.org>
Forwarded: no
Reviewed-By: Xavier Guimard <yadd@debian.org
Last-Update: 2020-02-10

--- a/master/master.c
+++ b/master/master.c
@@ -1968,7 +1968,7 @@
     struct rlimit rl;
 
 #ifdef HAVE_GETRLIMIT
-    if (!getrlimit(RLIMIT_NUMFDS, &rl)) {
+    if (getrlimit(RLIMIT_NUMFDS, &rl) >= 0) {
         if (x != RLIM_INFINITY && rl.rlim_max != RLIM_INFINITY && x > rl.rlim_max) {
             syslog(LOG_WARNING,
                    "limit_fds: requested %" PRIu64 ", but capped to %" PRIu64,
@@ -1987,7 +1987,7 @@
                rl.rlim_cur, rl.rlim_max);
     }
 
-    if (setrlimit(RLIMIT_NUMFDS, &rl) < 0) {
+    if (setrlimit(RLIMIT_NUMFDS, &rl) < 0 && x != RLIM_INFINITY) {
         syslog(LOG_ERR,
                "setrlimit: Unable to set file descriptors limit to %ld: %m",
                rl.rlim_cur);
